### PR TITLE
Github Actions Pipeline for building and testing the extension

### PR DESCRIPTION
**Pull request by Zbigor**
Together with @SalvatoreZagaria 
- Both unit tests and e2e tests are run inside docker containers
- E2E Tests are run in parallel for vc23 and vc24
  - vc24 E2E tests take longer since they include the tests for coded test workflows
- E2E tests now stop on failure
- Unit test coverage info is uploaded to and can be viewed on `codecov.io`
- Github Actions job summary includes download links for 
  - vsix file
  - coverage info
  - error screenshots 
- E2E Error messages and their locations are parsed from the logs and are shown in the summary

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -26,11 +26,45 @@
...
+      "console": "integratedTerminal"
+    },
+    {
+      "name": "getCompletion",
+      "type": "python",
+      "request": "launch",
+      "program": "${workspaceFolder}/python/testEditorInterface.py",
+      "cwd": "C:/RDS/VectorCAST/testExplorer/unitTests",
+      "env": {
+        "PYTHONPATH": "C:/vcast/vc23/python"
```
Comment (aytey) [inline .vscode/launch.json:59]: What's with these changes?

File: src/vcastTestInterface.ts [hunk 2]
```diff
@@ -902,4 +902,4 @@ export async function openCodedTest(testNode: testNodeType) {
   if (fs.existsSync(testNode.testFile)) {
     openFileWithLineSelected(testNode.testFile, testNode.testStartLine - 1);
   }
-}
+}
```
Comment (aytey) [inline src/vcastTestInterface.ts:905]: I think we can revert this?

File: .vscode/launch.json (hunk 1 above)
Comment (Zbigor) [reply .vscode/launch.json:59]: These were removed in ea4948a8e6419375b1d8b874586b4a4f313eca86 . They ended up here when I was branching from a stable version and dealing with merge conflicts. As they seemed to be unused commands now, I can get rid of these

File: src/vcastTestInterface.ts (hunk 2 above)
Comment (Zbigor) [reply src/vcastTestInterface.ts:905]: This a new line at the end of file, but yeah, we can revert if you want

File: python/vTestInterface.py [hunk 3]
```diff
@@ -588,3 +588,4 @@ def main():
         returnCode = 1
 
     sys.exit(returnCode)
+
```
[the remark this answers is not in the document]
Comment (Zbigor) [reply python/vTestInterface.py:591]: same as above

File: src/vcastTestInterface.ts (hunk 2 above)
Comment (aytey) [reply src/vcastTestInterface.ts:905]: Just to make it cleaner on the diff.

Then we can be sure there's no "application" changes in this.

File: python/vTestInterface.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aytey) [reply python/vTestInterface.py:591]: Just to make it cleaner on the diff.

Then we can be sure there's no "application" changes in this.

File: .vscode/launch.json (hunk 1 above)
Comment (aytey) [reply .vscode/launch.json:59]: Ah, yeah, would be good to remove these

**Pull request merged**